### ISSUE TITLE
Rule Change: (no-restricted-imports) Allow side-effects imports

### ISSUE THREAD

**Issue by AndreaPontrandolfo** [eslint/eslint#18991]
### What rule do you want to change?

no-restricted-imports

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

Before:

`import { x } from "node_modules/package/dist/x.js"` --> ✔️ this gets reported!

`import "node_modules/package/dist/x.js"` --> ❌ this gets reported!

After:

`import { x } from "node_modules/package/dist/x.js"` --> ✔️ this gets reported!

`import "node_modules/package/dist/x.js"` --> ✔️ this DOES NOT get reported!

### What does the rule currently do for this code?

I tried to use `no-restricted-imports` to restrict developers from importing directly from node_modules and dists folders (this can sometimes happen especially if the developer is using some sort of editor auto-import feature, then it gets overlooked in the code-review and it slips into production code).

```
    'no-restricted-imports': [
      2,
      {
        patterns: [
          {
            group: ['node_modules'],
            message: 'Imports from node_modules are likely a user mistake.',
          },
          {
            group: ['dist'],
            message: 'Imports from dist are likely a user mistake.',
          },
        ],
      },
    ],
```

I feel like this is fairly basic/common usecase for this rule.
However, i ran into the problem that there are some libraries that require to import code as side-effects, like css file:

`import "node_modules/package/dist/styles.css"`

Now, i obviously want to allow this kind of imports, but i didn't find to way to configure the rule to do so, given the current options.

### What will the rule do after it's changed?

Given my above configuration of the rule, this code should not trigger a warning:

`import "node_modules/package/dist/styles.css"`

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Maybe this could be accomplished with a opt-in option? Like:

```
    'no-restricted-imports': [
      2,
      {
        patterns: [
          {
            group: ['node_modules'],
            message: 'Imports from node_modules are likely a user mistake.',
            sideEffects: false // true by default
          },
          {
            group: ['dist'],
            message: 'Imports from dist are likely a user mistake.',
          },
        ],
      },
    ],
```

**Comment by mdjermanovic** [comment]
I think this can already be achieved by setting the existing option [`importNamePattern`](https://eslint.org/docs/latest/rules/no-restricted-imports#importnamepattern) to a pattern that matches any name, for example `importNamePattern: "."`.

Here's a [Playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgICduby1yZXN0cmljdGVkLWltcG9ydHMnOiBbXG4gICAgICAyLFxuICAgICAge1xuICAgICAgICBwYXR0ZXJuczogW1xuICAgICAgICAgIHtcbiAgICAgICAgICAgIGltcG9ydE5hbWVQYXR0ZXJuOiBcIi5cIixcbiAgICAgICAgICAgIGdyb3VwOiBbJ25vZGVfbW9kdWxlcyddLFxuICAgICAgICAgICAgbWVzc2FnZTogJ0ltcG9ydHMgZnJvbSBub2RlX21vZHVsZXMgYXJlIGxpa2VseSBhIHVzZXIgbWlzdGFrZS4nLFxuICAgICAgICAgIH0sXG4gICAgICAgICAge1xuICAgICAgICAgICAgaW1wb3J0TmFtZVBhdHRlcm46IFwiLlwiLFxuICAgICAgICAgICAgZ3JvdXA6IFsnZGlzdCddLFxuICAgICAgICAgICAgbWVzc2FnZTogJ0ltcG9ydHMgZnJvbSBkaXN0IGFyZSBsaWtlbHkgYSB1c2VyIG1pc3Rha2UuJyxcbiAgICAgICAgICB9LFxuICAgICAgICBdLFxuICAgICAgfSxcbiAgICBdLCovXG5cbmltcG9ydCB7IHggfSBmcm9tIFwibm9kZV9tb2R1bGVzL3BhY2thZ2UvZGlzdC94LmpzXCI7IC8vIHRoaXMgZ2V0cyByZXBvcnRlZFxuXG5pbXBvcnQgXCJub2RlX21vZHVsZXMvcGFja2FnZS9kaXN0L3guanNcIjsgLy8gdGhpcyBET0VTIE5PVCBnZXQgcmVwb3J0ZWQhXG5cbiIsIm9wdGlvbnMiOnsicnVsZXMiOnt9LCJsYW5ndWFnZU9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSIsInBhcnNlck9wdGlvbnMiOnsiZWNtYUZlYXR1cmVzIjp7fX19fX0=) with this config and your examples. Would this work for your use case?

**Comment by AndreaPontrandolfo** [comment]
Oh thank you! It works perfectly.

It might be valuable for other users to add this to the docs?

**Comment by mdjermanovic** [comment]
> It might be valuable for other users to add this to the docs?

Sounds good to me, I added an example in https://github.com/eslint/eslint/pull/18997.
